### PR TITLE
Added accessor methods for ANN layers.

### PR DESCRIPTION
**Pull request by iamshnoo**
This PR adds accessor methods for different layer parameters. The tests are present in the ann_layer_tests.cpp file and all of them have a similarity in their names (each one ends with LayerParametersTest).

Note : After messing up my PR in #2307, I deleted my fork and re-did everything in this new forked branch. All the changes are still the same. And I am yet to complete writing tests for this. 
This solves Issue #2258 temporarily to some extent for now.

### CODE REVIEW

**Comment by zoq** [comment]
Can you merge the current master branch?

**Comment by iamshnoo** [comment]
Done!

**Comment by zoq** [comment]
This PR is marked as a draft, but I think it's ready?

**Comment by iamshnoo** [comment]
The code is pretty much ready. I just kept it as a draft because I haven't written any test cases yet for the functions I added. Anyway, I am marking it as ready now. Let me know if there are any changes to make.

**Comment by zoq** [comment]
Thanks for the clarification, adding tests would be great!

**Comment by iamshnoo** [comment]
I will get down to writing the tests as soon as I can, maybe in a day or two. And then maybe, you can review it? Thanks!

**Comment by zoq** [comment]
Sounds good, no hurry.

**Comment by iamshnoo** [comment]
Closed this PR temporarily to make some changes in the branch. Will reopen very soon (hopefully)!

**Comment by iamshnoo** [comment]
I had to close and re-open this because of a tiny error I made when doing a git reset.

**Comment by zoq** [comment]
Just resolved the merge conflict, so make sure to run `git pull`.

File: src/mlpack/tests/ann_layer_test.cpp [hunk 1]
```diff
@@ -166,6 +179,19 @@ BOOST_AUTO_TEST_CASE(JacobianConstantLayerTest)
   }
 }
 
+/**
+ * Test that the function that can access the outSize parameter of the
+ * Constant layer works.
+ */
+BOOST_AUTO_TEST_CASE(ConstantLayerParametersTest)
```
Comment (zoq) [inline src/mlpack/tests/ann_layer_test.cpp:186]: Thanks for adding the tests!

**Comment by zoq** [comment]
Thanks again for the contribution.

**Pull request merged**